app/views/game_control/event_locations/_form.html.slim

```diff
@@ -18,6 +18,16 @@
                           input_html: { class: :state }
     = f.input :addr_postal_code
     = f.input :notes
+  - unless nil == @recent_locations || @recent_locations.empty?
+    .box-body
+      h3 Copy a Previous Location...
+      - @recent_locations.each do |recent_location|
+        table#previousLocations
+          tr
+            td.col-md-1
+              <input type="button" name="Copy" value="Copy" />
```

Comment:
You do not need tags in slim you can just do

`input type="button" name="copy" value="copy"`

For links you can use [`link_to` helper](http://api.rubyonrails.org/v4.2.7.1/classes/ActionView/Helpers/UrlHelper.html#method-i-link_to) which allows you to define HTTP verbs. 

I would add a PATCH action to the controller that copies a previous event_location into the current one.

You would need to add an action in the `config/routes/rb` file around line 30.

```ruby
resources 'event_locations' do
  patch 'copy'
end
```

That creates a URL helper path `game_control_event_event_location_copy`. (To see all the paths, `bin/rake routes`).

In the view you can replace the link with:

`link_to 'Use this location', game_control_event_event_location_copy_path(@location.event, @location, source_location: recent_location.id), method: :patch`

In the controller, add a method called `copy` that creates a duplicate of that selected event.  Inspect the var `params` that is available to methods in controller so see how the data is passed in.

To make it look like a button you can style it [like here](https://github.com/PuzzledPint/puzzledpint.com/blob/master/app/views/game_control/cities/index.html.slim#L12).

We are using [AdminLTE](https://almsaeedstudio.com/themes/AdminLTE/index.html) as the skin for the admin. Can you find [button classes here](https://almsaeedstudio.com/themes/AdminLTE/pages/UI/buttons.html).